### ISSUE TITLE
fdtransfer with net=host setting

### ISSUE THREAD

**Issue by Jongy** [jvm-profiling-tools/async-profiler#634]
fdtransfer creates a UDS socket in the net NS of the target program. The UDS is per PID. This has the hidden assumption that same PIDs (i.e different processes in different PID namespaces that share the same NSPID) do not share network namespaces.

If 2 processes do share network NS (e.g 2 Java containers running with --net=host parameter) - fdtransfer will run in the shared netns but it will actually target only one of the processes (because it is pairted with the real PID, not NSPID).

I have this case on a real deployment so I'm thinking what can be done.

An idea I got now while writing this post (@apangin let me know what you think about it, I can open a PR) is to include the `/proc/pid/ns/pid` inode (the number you get when you `readlink` this file). It is guaranteed to be different between different PID namespaces. Unprivileged processes can readlink their `ns/pid` and get the number, and fdtransfer can do so as well - together they provide a unique identifier across the system. This way, no matter if the net namespace is shared or not, it will be unique.

**Comment by apangin** [comment]
The proposed solution may work, but are you sure it's universally applicable to all supported Linux flavors?
What if `/proc/self/ns/pid` does not exist or is not accessible inside a container?

On the other hand, the socket name does not need to relate to `pid` at all. It can be any unique string, as long as we pass the very same string both to `fdtransfer` and to the target process (as an agent argument).

**Comment by Jongy** [comment]
> The proposed solution may work, but are you sure it's universally applicable to all supported Linux flavors? What if `/proc/self/ns/pid` does not exist or is not accessible inside a container?

If `/proc/self/ns/pid` doesn't exist - I don't read the number at all (and use 0) which makes sense because if no PID namespaces exist, then we don't need this number of uniqueness. I implemented in https://github.com/Jongy/async-profiler/commit/12e079989e85080e1ec00781d5ba3dcef320cf51.
AFAIK this will always be present in containers (its' under standard procfs), and the process' PID NS is always accessible for itself to read.

But, in any case ...

> On the other hand, the socket name does not need to relate to `pid` at all. It can be any unique string, as long as we pass the very same string both to `fdtransfer` and to the target process (as an agent argument).

I realized that as well while I was writing the code. This requires more changes in the code but will keep the API with OS simpler and will also ensure fdtransfers of different `profiler.sh` invocations no longer need fight for the same socket path (so in case an fdtransfer program remains hanging, it doesn't affect subsequent runs of async-profiler).

That solution makes more sense, I'll implement that
